src/dstore/pmix_esh.c

```diff
@@ -2107,9 +2110,8 @@ static int put_empty_ext_slot(seg_desc_t *dataseg)
 
 static size_t put_data_to_the_end(ns_track_elem_t *ns_info, seg_desc_t *dataseg, char *key, void *buffer, size_t size)
 {
-    size_t offset;
+    size_t offset, id = 0;
```

Comment:
so you say that integer was overflowed?
segment size if 4K by default. To overflow it you need to store 2G*4K which is unlikely I hope.